bokehjs/src/coffee/models/mappers/categorical_color_mapper.coffee

```diff
@@ -1,22 +1,56 @@
+import {ColorMapper} from "./color_mapper"
+
 import * as p from "core/properties"
+import {isArray} from "core/util/types"
 
-import {ColorMapper} from "./color_mapper"
+_equals = (a, b) ->
+  if a.length != b.length
+    return false
+  for i in [0...a.length]
+    if a[i] != b[i]
+      return false
+  return true
+
+_find = (a, b) ->
```

Comment:
Thanks @mattpap I have updated and added full tests for the categorical colormapper as well